src/hooks/useApi.js

```diff
@@ -0,0 +1,87 @@
+import axios from "axios";
+import { useCallback } from "react";
+import useSWR from "swr";
+
+/**
+ * @param {string} url - excluding base, i.e. /api/v1/users` should just be `/users`.
+ * @param {import("axios").AxiosRequestConfig} options - axios options.
+ * @returns
+ */
+export const useApi = (url, options) => {
+  const { data, error, ...rest } = useSWR(url, fetcher(options));
+  const loading = !data && !error;
+
+  return { data, error, ...rest, loading };
+};
+
+/**
+ * @param {string} url
+ * @param {import("axios").AxiosRequestConfig} options
+ *
+ */
+export const useMutation = (url, options) => {
+  const fn = useCallback(
+    /**
+     * @param {import("axios").AxiosRequestConfig} overrideOptions
+     */
+    (overrideOptions) => {
+      return fetcher({
+        ...{ ...options, ...overrideOptions },
+      })(url);
```

Comment:
Just wondering why we don't include url as a part of the parameter to the fetcher function?

Reply:
SWR passes the URL into the function returned by fetcher